src/net/sourceforge/kolmafia/Modifiers.java

```diff
@@ -2257,136 +2257,131 @@ public static final String trimModifiers(final String modifiers, final String re
+        this.set(Modifiers.HP_REGEN_MIN, dotw == DayOfWeek.SATURDAY ? 3.0 : 0.0);
+        this.set(Modifiers.HP_REGEN_MAX, dotw == DayOfWeek.SATURDAY ? 7.0 : 0.0);
+        return true;
+      }
+      case ItemPool.PANTSGIVING -> {
+        this.set(Modifiers.DROPS_ITEMS, Preferences.getInteger("_pantsgivingCrumbs") < 10);
+        return true;
+      }
+      case ItemPool.PATRIOT_SHIELD -> {
+        // Muscle classes
+        this.set(Modifiers.HP_REGEN_MIN, 0.0);
+        this.set(Modifiers.HP_REGEN_MAX, 0.0);
+        // Seal clubber
+        this.set(Modifiers.WEAPON_DAMAGE, 0.0);
+        this.set(Modifiers.DAMAGE_REDUCTION, 0.0);
+        // Turtle Tamer
+        this.set(Modifiers.FAMILIAR_WEIGHT, 0.0);
+        // Disco Bandit
+        this.set(Modifiers.RANGED_DAMAGE, 0.0);
+        // Accordion Thief
+        this.set(Modifiers.FOUR_SONGS, false);
+        // Mysticality classes
+        this.set(Modifiers.MP_REGEN_MIN, 0.0);
+        this.set(Modifiers.MP_REGEN_MAX, 0.0);
+        // Pastamancer
+        this.set(Modifiers.COMBAT_MANA_COST, 0.0);
+        // Sauceror
+        this.set(Modifiers.SPELL_DAMAGE, 0.0);
+
+        // Set modifiers depending on Character class
+        AscensionClass ascensionClass = KoLCharacter.getAscensionClass();
+        if (ascensionClass != null) {
+          switch (ascensionClass) {
+            case SEAL_CLUBBER, ZOMBIE_MASTER, ED, COWPUNCHER, BEANSLINGER, SNAKE_OILER -> {
+              this.set(Modifiers.HP_REGEN_MIN, 10.0);
+              this.set(Modifiers.HP_REGEN_MAX, 12.0);
+              this.set(Modifiers.WEAPON_DAMAGE, 15.0);
+              this.set(Modifiers.DAMAGE_REDUCTION, 1.0);
+            }
+            case TURTLE_TAMER -> {
+              this.set(Modifiers.HP_REGEN_MIN, 10.0);
+              this.set(Modifiers.HP_REGEN_MAX, 12.0);
+              this.set(Modifiers.FAMILIAR_WEIGHT, 5.0);
+            }
+            case DISCO_BANDIT, AVATAR_OF_SNEAKY_PETE -> this.set(Modifiers.RANGED_DAMAGE, 20.0);
+            case ACCORDION_THIEF -> this.set(Modifiers.FOUR_SONGS, true);
+            case PASTAMANCER -> {
+              this.set(Modifiers.MP_REGEN_MIN, 5.0);
+              this.set(Modifiers.MP_REGEN_MAX, 6.0);
+              this.set(Modifiers.COMBAT_MANA_COST, -3.0);
+            }
+            case SAUCEROR, AVATAR_OF_JARLSBERG -> {
+              this.set(Modifiers.MP_REGEN_MIN, 5.0);
+              this.set(Modifiers.MP_REGEN_MAX, 6.0);
+              this.set(Modifiers.SPELL_DAMAGE, 20.0);
             }
           }
-          return true;
         }
+        return true;
+      }
     }
     return false;
   }
 
   private boolean overrideSkill(final String name) {
-    switch (name) {
-      case "Ferocity":
-        if (KoLCharacter.isVampyre()) {
-          this.set(Modifiers.HP, -10.0);
-        } else if (KoLCharacter.isAvatarOfBoris()) {
-          this.set(Modifiers.CRITICAL_PCT, 25.0);
-        }
-        return true;
+    if ("Ferocity".equals(name)) {
```

Comment:
Why remove the switch?

It is true that there is only one case, but if another skill comes along, we'll just have to turn it back into a switch in order to add a second case.

Reply:
I let IntelliJ take the wheel here.

Though this reminds, me that it feels off. One of these is `[11002]Ferocity` and the other is `[24017]Ferocity`. Is there a way we can just maintain that disambiguation?

Reply:
Use those as the names? SkillDatabase.getSkillId() will accept a bracketed number as the skill id.

Reply:
I *think* it's a bit more complicated than that

Reply:
Yes I see - modifier lookups for skills don't deal in ids at all. I have a bit of code to write.

Reply:
What am I missing?
```
# Ferocity (Boris): Critical Hit Percent: +25
Skill	[11002]Ferocity	Critical Hit Percent: +25
# Ferocity (Vampyre): Your combat skills do 50% more damage
Skill	[24017]Ferocity	Spell Damage Percent: +50, Maximum HP: -10
```
yields:
```
> ash $skill[Ferocity].name

Multiple matches for "Ferocity"; using "[24017]Ferocity". (char 8 to char 16) Clarify by using one of:
$skill[[11002]Ferocity]
$skill[[24017]Ferocity]
Returned: Ferocity

> ash string_modifier($skill[[11002]Ferocity], "Modifiers")

Returned: Critical Hit Percent: +25

> ash string_modifier($skill[[24017]Ferocity], "Modifiers")

Returned: Spell Damage Percent: +50, Maximum HP: -10
```

Reply:
Actually having the skill will not match as a result

Reply:
My following commits fix the issue :)